src/main/java/com/coveo/spillway/storage/InMemoryStorage.java

```diff
@@ -53,25 +52,24 @@
 public class InMemoryStorage implements LimitUsageStorage {
 
   Map<Instant, Map<LimitKey, Capacity>> map = new ConcurrentHashMap<>();
-  private Object lock = new Object();
   private Clock clock = Clock.systemDefaultZone();
 
   @Override
   public Map<LimitKey, Integer> addAndGet(Collection<AddAndGetRequest> requests) {
-    Map<LimitKey, Integer> updatedEntries = new LinkedHashMap<>();
-    synchronized (lock) {
-      for (AddAndGetRequest request : requests) {
-        Instant expirationDate = request.getBucket().plus(request.getExpiration());
+    Map<LimitKey, Integer> updatedEntries = new HashMap<>();
 
-        LimitKey limitKey = LimitKey.fromRequest(request);
+    for (AddAndGetRequest request : requests) {
+      Instant expirationDate = request.getBucket().plus(request.getExpiration());
 
-        Map<LimitKey, Capacity> mapWithThisExpiration =
-            map.computeIfAbsent(expirationDate, (key) -> new HashMap<>());
-        Capacity counter = mapWithThisExpiration.computeIfAbsent(limitKey, (key) -> new Capacity());
-        updatedEntries.put(limitKey, counter.addAndGet(request.getCost()));
-      }
-      removeExpiredEntries();
+      LimitKey limitKey = LimitKey.fromRequest(request);
+
+      Map<LimitKey, Capacity> mapWithThisExpiration =
+          map.computeIfAbsent(expirationDate, (key) -> new HashMap<>());
```

Comment:
Yes I had not thought about it, but it makes sense.
If the hashmap already exists and two threads try to modify it, I guess it might cause some concurrency problem.
Is the switch to a ConcurrentHashMap more costly than keeping the Sync? I feel we should chose a method (concurrent or lock) and stock to it.